### ISSUE TITLE
setting entry in struct array with object field

### ISSUE THREAD

**Issue by anntzer** [numpy/numpy#6806]
```
In [23]: a = np.ones(3, 'i4, O')

In [24]: a[0] = a[1]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-24-5ef5a6077ce6> in <module>()
----> 1 a[0] = a[1]

ValueError: Setting void-array with object members using buffer.
```
numpy 1.10.1
First reported by @ldoddema in #6776.

**Comment by ahaldane** [comment]
This is an old bug present in 1.9.3 (or before). It happens because of the behavior of structured item assignemt (in `VOID_setitem`), which treats the lhs as a buffer. This causes a number of known bugs.

A possible fix to this is implemented in #6053. (See links there for related bugs)

**Comment by charris** [comment]
Closing as duplicate.
